base/broadcast.jl

```diff
@@ -1261,4 +1261,10 @@ end
 end
 @inline broadcasted(::S, f, args...) where S<:BroadcastStyle = Broadcasted{S}(f, args)
 
+struct BroadcastOp{F} <: Function
```

Comment:
Needs a documentation string.

Reply:
Does this need to be a struct?  It could just be:

```julia
broadcast_op(f) = (x...) -> f.(x...)
```

That's how, for example, the somewhat-equivalent `Base.splat` function works.  The downside is that you can no longer dispatch on it, I suppose.

This also works for non-operators, which makes me want for a better name.  Maybe `broadcasting`?  The gerund actually makes sense here, and I think `Base.splatting` might even make more sense than `Base.splat`.

Reply:
The reason I decided to make this a struct is primarily for better stack traces, which is, I believe, the reason we also have `Base.Fix2`, for example. It would also open up the opportunity for higher-order functions to specialize on `BroadcastOp`, i.e. to fuse it with other broadcasting operations.
I am certainly open to suggestions for a better name! Having `broadcasting` together with `broadcast` and `broadcasted` might be a little hard to remember and perhaps throw off some non-native speakers.

Reply:
A `struct` also allows much more informative pretty-printing, e.g. if someone types `julia> .*` in the REPL.

Reply:
> This also works for non-operators

So how about `BroadcastingFunction`?  Since it's unlikely that this is typed interactively, maybe something very descriptive makes sense?

Reply:
Or maybe just `BroadcastFunc`.

Reply:
Sounds good to me too.